crates/rpc/src/v07/dto/header.rs

```diff
@@ -0,0 +1,214 @@
+use crate::felt::RpcFelt;
+use pathfinder_common::prelude::*;
+use serde::{ser::SerializeStruct, Serialize, Serializer};
+use serde_with::serde_as;
+
+#[serde_as]
+#[derive(Serialize)]
+pub struct Header {
+    #[serde_with(as = "RpcFelt")]
+    block_hash: BlockHash,
+    #[serde_with(as = "RpcFelt")]
+    parent_hash: BlockHash,
+    block_number: BlockNumber,
+    #[serde_with(as = "RpcFelt")]
+    new_root: StateCommitment,
+    timestamp: BlockTimestamp,
+    #[serde_with(as = "RpcFelt")]
+    sequencer_address: SequencerAddress,
+    l1_gas_price: ResourcePrice,
+    starknet_version: StarknetVersion,
+    l1_data_gas_price: ResourcePrice,
+    l1_da_mode: L1DaMode,
+}
+
+impl From<pathfinder_common::BlockHeader> for Header {
+    fn from(value: pathfinder_common::BlockHeader) -> Self {
+        let l1_gas_price = ResourcePrice {
+            wei: value.eth_l1_gas_price,
+            fri: value.strk_l1_gas_price,
+        };
+        let l1_data_gas_price = ResourcePrice {
+            wei: value.eth_l1_data_gas_price,
+            fri: value.strk_l1_data_gas_price,
+        };
+
+        Self {
+            block_hash: value.hash,
+            parent_hash: value.parent_hash,
+            block_number: value.number,
+            new_root: value.state_commitment,
+            timestamp: value.timestamp,
+            sequencer_address: value.sequencer_address,
+            l1_gas_price,
+            starknet_version: value.starknet_version,
+            l1_data_gas_price,
+            l1_da_mode: value.l1_da_mode.into(),
+        }
+    }
+}
+
+#[serde_as]
+#[derive(Serialize)]
+pub struct PendingHeader {
+    #[serde_with(as = "RpcFelt")]
+    parent_hash: BlockHash,
+    timestamp: BlockTimestamp,
+    #[serde_with(as = "RpcFelt")]
+    sequencer_address: SequencerAddress,
+    l1_gas_price: ResourcePrice,
+    starknet_version: StarknetVersion,
+}
```

Comment:
I see that this matches the spec but the spec itself is missing L1 data gas price and `l1_da_mode`. Isn't that a spec oversight?

Reply:
Potentially. Will ask.